crates/lib3h/src/transport/memory_mock/ghost_transport_memory.rs

```diff
@@ -39,20 +44,55 @@ pub type GhostTransportMemoryEndpointContextParent = GhostContextEndpoint<
 
 #[allow(dead_code)]
 pub struct GhostTransportMemory {
+    machine_id: Address,
     network: String,
     endpoint_parent: Option<GhostTransportMemoryEndpoint>,
     endpoint_self: Detach<GhostTransportMemoryEndpointContext>,
     /// My peer uri on the network layer (not None after a bind)
     maybe_my_address: Option<Url>,
     /// Addresses of connections to remotes
     connections: HashSet<Url>,
+    last_discover: Option<Instant>,
+    discover_interval_ms: u128,
+}
+
+impl Discovery for GhostTransportMemory {
+    fn advertise(&mut self) -> DiscoveryResult<()> {
+        let uri = self
+            .maybe_my_address
+            .clone()
+            .ok_or_else(|| DiscoveryError::new_other("must bind before discovering"))?;
+        let mut verse = memory_server::get_memory_verse();
+        let _ = verse
+            .get_net(&self.network)
+            .ok_or_else(|| DiscoveryError::new_other("net not found"))?
+            .advertise(uri, self.machine_id.clone());
+        Ok(())
+    }
+    fn discover(&mut self) -> DiscoveryResult<Vec<Url>> {
+        let mut verse = memory_server::get_memory_verse();
+        let machines = verse
+            .get_net(&self.network)
+            .ok_or_else(|| DiscoveryError::new_other("net not found"))?
+            .discover();
+        Ok(machines.into_iter().map(|(uri, _)| uri).collect())
+    }
+    fn release(&mut self) -> DiscoveryResult<()> {
+        Ok(())
+    }
+    fn flush(&mut self) -> DiscoveryResult<()> {
+        Ok(())
+    }
 }
 
 impl GhostTransportMemory {
     #[allow(dead_code)]
-    pub fn new(network_name: &str) -> Self {
+    pub fn new(machine_id: Address, network_name: &str) -> Self {
         let (endpoint_parent, endpoint_self) = create_ghost_channel();
+        let interval = 30000;
```

Comment:
Eventually should b a config parameter of engine or at the very least make a well named constant for this?

Reply:
added